src/prefect/server/api/flow_runs.py

```diff
@@ -110,6 +112,61 @@ async def count_flow_runs(
         )
 
 
+@router.post("/lateness")
+async def average_flow_run_lateness(
+    flows: Optional[schemas.filters.FlowFilter] = None,
+    flow_runs: Optional[schemas.filters.FlowRunFilter] = None,
+    task_runs: Optional[schemas.filters.TaskRunFilter] = None,
+    deployments: Optional[schemas.filters.DeploymentFilter] = None,
+    work_pools: Optional[schemas.filters.WorkPoolFilter] = None,
+    work_pool_queues: Optional[schemas.filters.WorkQueueFilter] = None,
+    db: PrefectDBInterface = Depends(provide_database_interface),
+) -> float | None:
+    """
+    Query for average flow-run lateness in seconds.
+    """
+    async with db.session_context() as session:
+        query = await models.flow_runs._apply_flow_run_filters(
+            sa.select(
+                sa.func.avg(
+                    sa.case(
```

Comment:
Since we want an average of the lateness we're unable to use the existing `FlowRun.expected_start_time_delta` property as it returns a timedelta and SQLite is unable to properly deal with it always returning `1970.0`. This copies the same logic but ensures that it returns the number of seconds instead.

Reply:
I turned this into a comment so we have it along side the code.